src/string.cr

```diff
@@ -4344,11 +4558,17 @@ class String
     pointerof(@c)
   end
 
-  def unsafe_byte_slice(byte_offset, count)
+  # Returns *count* of underlying bytes of this String starting at given *byte_offset*.
+  #
+  # The returned slice is read-only.
+  def unsafe_byte_slice(byte_offset, count) : Slice
```

Comment:
Perhaps these should be `:nodoc:` instead?

They're unsafe because they don't do any bounds checking.

Reply:
I don't like undocumented public methods. It does not prevent from use them, only make them mysteriously hidden.. Please remove them or make clear their risks in doc.

Reply:
I'd leave this as is for now.